lib/puppetserver/ca/setup_command.rb

```diff
@@ -0,0 +1,136 @@
+require 'optparse'
+require 'puppetserver/ca/x509_loader'
+require 'puppetserver/ca/puppet_config'
+
+module Puppetserver
+  module Ca
+    class SetupCommand
+
+      def initialize(logger)
+        @logger = logger
+      end
+
+      def run!(input)
+        bundle_path = input['cert-bundle']
+        key_path = input['private-key']
+        chain_path = input['crl-chain']
+        config_path = input['config']
+
+        files = [bundle_path, key_path, chain_path, config_path].compact
+
+        errors = validate_file_paths(files)
+        return 1 if log_possible_errors(errors)
+
+        unless chain_path
+          @logger.err 'Warning:'
+          @logger.err '    No CRL chain given'
+          @logger.err '    Full CRL chain checking will not be possible'
+          @logger.err ''
+        end
+
+        loader = X509Loader.new(bundle_path, key_path, chain_path)
+        return 1 if log_possible_errors(loader.errors)
+
+        puppet = PuppetConfig.parse(config_path)
+        return 1 if log_possible_errors(puppet.errors)
+
+
+        File.open(puppet.settings[:cacert], 'w') do |f|
```

Comment:
This is just a reminder (Justin and I discussed):
We'll probably want to create a /puppetlabs/etc/puppet directory if there isn't one made yet.
When I blew away my .puppetlabs directory, things started to fail:
```
 rm -rf .puppetlabs
 ~ $🐰  cd .puppetlabs
-bash: cd: .puppetlabs: No such file or directory
 ~ $🐰  cd Development/puppetserver-ca-cli/
 ~/Development/puppetserver-ca-cli (SERVER-2172-wip) $🐰  bundle exec exe/puppetserver-ca setup --private-key=bin/privkey.pem --cert-bundle=bin/ca.pem
Warning:
    No CRL chain given
    Full CRL chain checking will not be possible

bundler: failed to load command: exe/puppetserver-ca (exe/puppetserver-ca)
Errno::ENOENT: No such file or directory @ rb_sysopen - /Users/amy.sahli/.puppetlabs/etc/puppet/ssl/ca/ca_crt.pem
  /Users/amy.sahli/Development/puppetserver-ca-cli/lib/puppetserver/ca/setup_command.rb:38:in `initialize'
  /Users/amy.sahli/Development/puppetserver-ca-cli/lib/puppetserver/ca/setup_command.rb:38:in `open'
  /Users/amy.sahli/Development/puppetserver-ca-cli/lib/puppetserver/ca/setup_command.rb:38:in `run!'
  /Users/amy.sahli/Development/puppetserver-ca-cli/lib/puppetserver/ca/cli.rb:39:in `run!'
  exe/puppetserver-ca:5:in `<top (required)>'
```